scanner/src/main/kotlin/PathScanner.kt

```diff
@@ -248,7 +248,7 @@ abstract class PathScanner(
      * Scan the provided [pkg] for license information and return a [ScanResult]. The package's source code is
      * downloaded to [downloadDirectory] for scanning. If the package could not be scanned a [ScanException] is thrown.
      */
-    fun scanPackage(scannerDetails: ScannerDetails, pkg: Package, downloadDirectory: File): ScanResult {
+    private fun scanPackage(scannerDetails: ScannerDetails, pkg: Package, downloadDirectory: File): ScanResult {
```

Comment:
I basically agree to the rationale provided in tat commit, but now that the filtering for ignored paths happens on a higher level, calling this function directly would yield wrong results.

If really needed, a way to solve this would be to rename this function to `scanPackageInternal` (or similar) and provide a new `scanPackage` function that calls `scanPackage` and does the filtering. But I wanted to avoid that, as we already have a confusing amount of scan functions, IMO.

Reply:
I think it's only used in the web app prototype I built a while ago and I do not plan to continue working on that. If it becomes relevant again, e.g. in the context of the ORT Workbench, we can adjust on demand.